Source/ARTDefault.m

```diff
@@ -6,9 +6,35 @@
 //  Copyright (c) 2015 Ably. All rights reserved.
 //
 
-#import "ARTDefault+Private.h"
-
 #import "Ably.h"
+#import "ARTDefault+Private.h"
+#import <sys/utsname.h>
+
+// NSOperatingSystemVersion has NSInteger as version components for some reason, so mitigate it here.
+static UInt32 conformVersionComponent(const NSInteger component) {
+    if (component < 0) return 0;
```

Comment:
Perhaps a little bit too complex. This implementation could perhaps be a one liner:

```objc
return (component < 0) ? 0 : (UInt32)component;
```

Though, if I'm honest, despite my initial direction that should make this a static function, it could be argued that a macro would also suffice.e.g.:

```objc
#define conformVersionComponent(component) ((component < 0) ? 0 : (Uint32)component)
```

Equally, happy to leave it as is - what you've got is still clear.

Reply:
https://github.com/ably/ably-cocoa/pull/1119/commits/890c069f75096e056d668f9d45f0300248ffc843